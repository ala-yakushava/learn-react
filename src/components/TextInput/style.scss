@import "../../styles/theme";

.TextInput {
  display: flex;
  flex-direction: column;

  + .TextInput {
    margin-top: 30px;
  }

  &_label {
    margin-bottom: 15px;
    color: $red;
    text-transform: uppercase;
  }

  &_field {
    width: 100%;
    padding: 17px 20px;
    font-size: 20px;
    color: $white;
    border-radius: 5px;
    border: none;
    background-color: $dark-grey;

    &:read-only {
      padding: 0;
      color: $white;
      background-color: transparent;
    }

    &::placeholder {
      color: rgb(155, 152, 152);
    }
  }
}
