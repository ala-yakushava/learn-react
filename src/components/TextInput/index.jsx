import React from 'react';
import PropTypes from 'prop-types';

import './style.scss';

const TextInput = ({
  label, name, value, placeholder, required, readOnly, onChange,
}) => (
  <div className="TextInput">
    {label && <label className="TextInput_label" htmlFor={name}>{label}</label>}
    <input
      id={name}
      name={name}
      className="TextInput_field"
      required={required}
      readOnly={readOnly}
      placeholder={placeholder}
      value={value}
      onChange={onChange}
    />
  </div>
);

TextInput.propTypes = {
  label: PropTypes.string,
  name: PropTypes.string,
  value: PropTypes.string,
  placeholder: PropTypes.string,
  required: PropTypes.bool,
  readOnly: PropTypes.bool,
  onChange: PropTypes.func,
};

TextInput.defaultProps = {
  label: null,
  name: '',
  value: '',
  placeholder: null,
  required: false,
  readOnly: false,
  onChange: () => {},
};

export default TextInput;
