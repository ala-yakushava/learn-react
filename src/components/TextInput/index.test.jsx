import React from 'react';
import renderer from 'react-test-renderer';

import TextInput from './index';

describe('Components - TextInput', () => {
  test('should match snapshot', () => {
    const tree = renderer
      .create(<TextInput />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
