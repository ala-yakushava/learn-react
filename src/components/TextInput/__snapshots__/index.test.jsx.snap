// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - TextInput should match snapshot 1`] = `
<div
  className="TextInput"
>
  <input
    className="TextInput_field"
    id=""
    name=""
    onChange={[Function]}
    placeholder={null}
    readOnly={false}
    required={false}
    type="text"
    value=""
  />
</div>
`;
